barwidget.go

```diff
@@ -0,0 +1,178 @@
+package desktop
+
+import (
+	"image/color"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/driver/desktop"
+	"fyne.io/fyne/layout"
+	"fyne.io/fyne/theme"
+	"fyne.io/fyne/widget"
+)
+
+//Bar is the main widget housing the icon launcher
+type Bar struct {
+	baseWidget
+
+	Children      []fyne.CanvasObject
+	MouseInside   bool
+	MousePosition fyne.Position
+}
+
+//MouseIn alerts the widget that the mouse has entered
+func (b *Bar) MouseIn(*desktop.MouseEvent) {
+	b.MouseInside = true
+}
+
+//MouseOut alerts the widget that the mouse has left
+func (b *Bar) MouseOut() {
+	b.MouseInside = false
+	widget.Renderer(b).Layout(b.Size())
+}
+
+//MouseMoved alerts the widget that the mouse has changed position
+func (b *Bar) MouseMoved(event *desktop.MouseEvent) {
+	b.MousePosition = event.Position
+	widget.Renderer(b).Layout(b.Size())
+}
+
+//Resize resizes the widget to the provided size
+func (b *Bar) Resize(size fyne.Size) {
+	b.resize(size, b)
+}
+
+//Move moves the widget to the provide position
+func (b *Bar) Move(pos fyne.Position) {
+	b.move(pos, b)
+}
+
+//MinSize returns the minimum size of the widget
+func (b *Bar) MinSize() fyne.Size {
+	return b.minSize(b)
+}
+
+//Show makes the widget visible
+func (b *Bar) Show() {
+	b.show(b)
+}
+
+//Hide makes the widget hidden
+func (b *Bar) Hide() {
+	b.hide(b)
+}
+
+//Prepend adds an object to the begging of the widget
+func (b *Bar) Prepend(object fyne.CanvasObject) {
+	if b.Hidden && object.Visible() {
+		object.Hide()
+	}
+	b.Children = append([]fyne.CanvasObject{object}, b.Children...)
+
+	widget.Refresh(b)
+}
+
+//Append adds an object to the end of the widget
+func (b *Bar) Append(object fyne.CanvasObject) {
+	if b.Hidden && object.Visible() {
+		object.Hide()
+	}
+	b.Children = append(b.Children, object)
+
+	widget.Refresh(b)
+}
+
+//AppendSeparator adds a separator between the default icons and the taskbar
+func (b *Bar) AppendSeparator() {
+	object := canvas.NewRectangle(theme.TextColor())
+	if b.Hidden && object.Visible() {
+		object.Hide()
+	}
+	b.Children = append(b.Children, object)
+
+	widget.Refresh(b)
+}
+
+//AppendTaskbar adds an object to the taskbar area of the widget just before the final spacer
+func (b *Bar) AppendTaskbar(object fyne.CanvasObject) {
```

Comment:
This seems like a slightly misleading name as it's not appending a taskbar - perhaps "appendToTaskbar" is better?